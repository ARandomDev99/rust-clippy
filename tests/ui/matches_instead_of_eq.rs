#![warn(clippy::matches_instead_of_eq)]

enum Foo {
    Bar,
    Baz,
}

#[derive(PartialEq)]
enum EvenOdd {
    Even,
    Odd,
    Unknown,
}

#[derive(PartialEq)]
enum Foo2 {
    Bar(u8, u8),
    Baz { x: u8, y: u8 },
}

fn main() {
    let x = Foo::Bar;
    let val = matches!(x, Foo::Bar); // No error as Foo does not implement PartialEq

    let x = EvenOdd::Even;
    let val = matches!(x, EvenOdd::Even);
    //~^ matches_instead_of_eq

    let x = EvenOdd::Odd;
    let val = matches!(x, EvenOdd::Even | EvenOdd::Odd); // No error

    let x = Foo2::Bar(1, 2);
    let val = matches!(x, Foo2::Bar(_, _)); // No error

    let x = Foo2::Baz { x: 1, y: 2 };
    let val = matches!(x, Foo2::Baz { .. }); // No Error

    let val = matches!(x, Foo2::Bar(..) | Foo2::Baz { .. }); // No error
}
