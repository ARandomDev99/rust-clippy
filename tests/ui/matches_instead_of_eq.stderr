error: This expression can be replaced with a == comparison
  --> tests/ui/matches_instead_of_eq.rs:26:15
   |
LL |     let val = matches!(x, EvenOdd::Even);
   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: This type implements PartialEq, so you can use == instead of matches!()
   = note: `-D clippy::matches-instead-of-eq` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::matches_instead_of_eq)]`
   = note: this error originates in the macro `matches` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 1 previous error

